### ISSUE TITLE
Bot couldn't send messages and I get a Failed to spawn ngrokResponse code 502 (Bad Gateway)

### ISSUE THREAD

**Issue by anamariakantar** [Microsoft/BotFramework-Emulator#720]
Hi there,

I tried following the instructions here as I am developing with Python:
https://docs.microsoft.com/en-us/azure/bot-service/python/bot-builder-python-quickstart?view=azure-bot-service-4.0

 I get the bot listening on my port  (9002) but localhost:9002 comes with this error:

Error response
Error code: 501

Message: Unsupported method ('GET').

I added the endpoint of the bot at this URL: http://localhost:9002/api/messages

Error code explanation: HTTPStatus.NOT_IMPLEMENTED - Server does not support this operation.

I also tried setting up ngrok and I use the ngrok http 80 and edited the Emulator settings to the path of ngrok.exe

This is how the bot emulator interface looks like:

![image](https://user-images.githubusercontent.com/25516632/43493798-ecda3924-9583-11e8-9b6a-d4105d819012.png)

Can anyone help me out please?

Thanks,
Ana

**Comment by tonyanziano** [comment]
Hi @anamariakantar ,

Can you try contacting your bot without setting up ngrok, as it shouldn't be necessary if you are trying to contact a locally running bot.

Before sending a message to your bot, try setting a breakpoint inside of your `__handle_message_activity()` method, and then send the message. See if the breakpoint is hit and if you can step through the code to see if an exception is being thrown that results in the 501.

**Comment by anamariakantar** [comment]
Hi @tonyanziano ,

Thanks a lot for your reply. I tried debugging in the __handle_message_activity and it doesn't hit the breakpoint. I also tried stepping in the code and I get a key error at http.server - I am not sure why.

I also tried downloading the python samples from github and loaded the echobot with state, which works fine, so I am not sure what's going on. The code for this one uses botbuilder instead of botframework so that might explain some things?

I with my oo programming knowledge was better to figure out the issue. :(
